### PR TITLE
Add signal handler to close connection to iRODS before exiting

### PR DESCRIPTION
**Pull request by mksanger**
Catches SIGINT, SIGQUIT, SIGHUB, SIGTERM, SIGSEGV and SIGBUS.

### CODE REVIEW

File: src/signal_handler.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  case SIGTERM:
+  case SIGUSR1:
+  case SIGUSR2:
+    exit_flag = 3;
+    break;
+  case SIGPIPE:
+    exit_flag = 4;
+    break;
+  default:
+    exit_flag = 1;
```
Comment (kjsanger) [inline src/signal_handler.c:44]: There are a couple of signals we overlooked where the default behaviour is to stop / continue or to ignore the signal, so I don't think we should add behaviour changing that. i.e. SIGSTP, SIGSTOP / SIGCONT and SIGCHLD.

**Review by kjsanger — CHANGES_REQUESTED** [review]

File: src/signal_handler.c (hunk 1 above)
Comment (mksanger) [reply src/signal_handler.c:44]: What are you suggesting?  Only signals included in the `signals` array will have their action changed (see L58).  There are a couple of changes that I will make: SIGQUIT should have it's own case and SIGUSR1/2 should be in `signals`.  Is there anything I have missed?

Comment (kjsanger) [reply src/signal_handler.c:50]: Sorry - I'm trying to do too many things at once. Those signals are not in the array, so they won't be caught. Please ignore my comment!

Comment (kjsanger) [reply src/signal_handler.c:50]: And yes, I agree with your other amendments that I didn't spot.

**Pull request merged**